src/main/kotlin/com/cultureamp/eventsourcing/BatchedAsyncEventProcessor.kt

```diff
@@ -23,7 +23,8 @@ class BatchedAsyncEventProcessor<M : EventMetadata>(
         if (count > 0 || Random.nextFloat() < 0.01) {
             System.out.println("Finished processing batch for ${bookmark.name}, $count events up to sequence ${bookmark.sequence}")
         }
-    }
+    },
+    private val upcasting: Boolean = false,
```

Comment:
Kestrel is still young, I say we boldly enable this by default
```suggestion
    private val upcasting: Boolean = true,
```